src/androidTest/java/de/dennisguse/opentracks/services/TrackRecordingServiceTestStateMachine.java

```diff
@@ -101,13 +101,14 @@ public void tearDown() throws TimeoutException {
 
     @MediumTest
     @Test
-    public void initialState() {
+    public void initialState() throws InterruptedException {
         // given
         List<Track> tracks = contentProviderUtils.getTracks();
         assertTrue(tracks.isEmpty());
 
         // when
         // noop
+        Thread.sleep(1000);
```

Comment:
This should not be necessary; the other Thread.sleep()s are necessary as we check the observables and those are executed in a different thread (I presume).
isRecording() uses the value object directly, which is set in the same thread.

Reply:
This was an attempt to fix the assert with NOT_RECORDING 3 lines down. I'll remove it.